Userland/Libraries/LibC/malloc.cpp

```diff
@@ -198,8 +198,16 @@ enum class CallerWillInitializeMemory {
     Yes,
 };
 
+#ifndef NO_TLS
+thread_local
+#endif
+    bool s_allocation_enabled
+    = true;
+
 static void* malloc_impl(size_t size, CallerWillInitializeMemory caller_will_initialize_memory)
 {
+    VERIFY(s_allocation_enabled);
```

Comment:
Assuming we do want this feature, we should also verify no calls to `free()`?

Reply:
Sounds like a good idea